src/pudl/io_managers.py

```diff
@@ -0,0 +1,322 @@
+                        parent_table=parent_name,
+                        foreign_key=parent_fk,
+                        rowids=parent_fk_df["rowid"].values,
+                    )
+                )
+            raise ForeignKeyErrors(errors)
+
+    def _handle_pandas_output(self, context: OutputContext, df: pd.DataFrame):
+        """Write dataframe to the database."""
+        table_name = self._get_table_name(context)
+
+        sa_table = self._get_sqlalchemy_table(table_name)
+        engine = self.engine
+
+        # TODO (bendnorman) I included this if else statement for the analysis table example.
+        if sa_table is None:
+            with engine.connect() as con:
+                # Remove old table records before loading to db
+                df.to_sql(
+                    table_name,
+                    con,
+                    if_exists="replace",
+                    index=False,
+                )
+        else:
+            with engine.connect() as con:
+                # Remove old table records before loading to db
+                con.execute(sa_table.delete())
+
+                df.to_sql(
+                    table_name,
+                    con,
+                    if_exists="append",
+                    index=False,
+                    dtype={c.name: c.type for c in sa_table.columns},
+                )
+
+    def _handle_str_output(self, context: OutputContext, query: str):
+        """Execute a sql query on the database."""
+        engine = self.engine
+        table_name = self._get_table_name(context)
+
+        with engine.connect() as con:
+            # Drop the existing view if it exists and create the new view.
+            # TODO (bendnorman): parameterize this safely.
+            con.execute(f"DROP VIEW IF EXISTS {table_name}")
+            con.execute(query)
+
+    def handle_output(self, context: OutputContext, obj: pd.DataFrame | str):
+        """Handle an op or asset output."""
+        if isinstance(obj, pd.DataFrame):
+            self._handle_pandas_output(context, obj)
+        elif isinstance(obj, str):
+            self._handle_str_output(context, obj)
+        else:
+            raise Exception(
+                "SQLiteIOManager only supports pandas DataFrames and strings of SQL queries."
+            )
+
+    def load_input(self, context: InputContext) -> pd.DataFrame:
+        """Load a dataframe from a sqlite database."""
+        table_name = self._get_table_name(context)
+        _ = self._get_sqlalchemy_table(table_name)
+
+        engine = self.engine
+
+        with engine.connect() as con:
+            return pudl.metadata.fields.apply_pudl_dtypes(
+                pd.read_sql_table(table_name, con)
+            )
+
+
+@io_manager(
+    config_schema={
+        "pudl_output_path": Field(
+            str,
+            description="Path of directory to store the database in.",
+            default_value=os.environ.get("PUDL_OUTPUT"),
```

Comment:
I agree that this seems more approachable to users than the pudl_settings object. 

An error here if those vars aren't set would be useful, as I tried to import `pudl` without setting them and got an error about this Field.

Reply:
I resolved this issue by creating an EnvVar config type ([See previous commit](https://github.com/catalyst-cooperative/pudl/blob/b79b37fae3a47550e2ba709597ff6e3e0046a4c1/src/pudl/helpers.py#L1551-L1581)). However I reverted back to using the `pudl_settings` object because the `PudlTabl`, our examples and documentation expect it. I think it makes to deprecate the pudl_settings object once all of PUDL has been converted to use dagster.